app/src/main/java/org/literacyapp/authentication/AuthenticationActivity.java

```diff
@@ -0,0 +1,179 @@
+                    numberOfTries++;
+                    if (student != null){
+                        new StudentUpdateHelper(getApplicationContext(), student).updateStudent();
+                        finish();
+                    } else if (numberOfTries >= NUMBER_OF_MAXIMUM_TRIES){
+                        startActivity(new Intent(getApplicationContext(), StudentImageCollectionActivity.class));
+                    }
+                }
+            }
+        }
+
+        // Add overlay
+        animalOverlayHelper.addOverlay(imgRgba);
+
+        return imgRgba;
+    }
+
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+        TrainingHelper trainingHelper = new TrainingHelper(getApplicationContext());
+        svm = trainingHelper.getSvm();
+        tensorFlow = trainingHelper.getInitializedTensorFlow();
+        ppF = new PreProcessorFactory(getApplicationContext());
+        animalOverlayHelper.createOverlay();
+        numberOfTries = 0;
+        preview.enableView();
+    }
+
+    /**
+     * Returns the SVM string using the feature vector
+     * @param img
+     * @return
+     */
+    private synchronized String getSvmString(Mat img){
+        Mat featureVector = tensorFlow.getFeatureVector(img);
+        return svm.getSvmString(featureVector);
+    }
+
+    /**
+     * Determines if and which Student has been recognized. The Student is only returned if the probability is above 90%
+     * @param svmProbability
+     * @return
+     */
+    private synchronized Student getStudentFromProbability(String svmProbability){
+        // Example string for svmProbability: labels 1 2\n2 0.458817 0.541183
+        StringTokenizer stringTokenizerSvmProbability = new StringTokenizer(svmProbability, "\n");
+        // Example string for header: labels 1 2
+        String header = stringTokenizerSvmProbability.nextToken();
+        // Example string for content: 2 0.458817 0.541183
+        String content = stringTokenizerSvmProbability.nextToken();
+
+        StringTokenizer stringTokenizerHeader = new StringTokenizer(header, " ");
+        // Skip first token
+        stringTokenizerHeader.nextToken();
+        StringTokenizer stringTokenizerContent = new StringTokenizer(content, " ");
+        // First token shows the label with the highest probability
+        int labelWithHighestProbability = Integer.valueOf(stringTokenizerContent.nextToken());
+
+        HashMap<Integer, Double> probabilityMap = new HashMap<Integer, Double>();
+        while(stringTokenizerHeader.hasMoreTokens()){
+            probabilityMap.put(Integer.valueOf(stringTokenizerHeader.nextToken()), Double.valueOf(stringTokenizerContent.nextToken()));
+        }
+
+        if (probabilityMap.get(labelWithHighestProbability) > 0.9){
+            StudentImageCollectionEvent studentImageCollectionEvent = studentImageCollectionEventDao.load((long) labelWithHighestProbability);
```

Comment:
Is the label the id of a StudentImageCollectionEvent? If so, could we name it something like "eventIdWithHighestProbability" instead?